cli/src/main/java/org/eclipse/hono/cli/client/Client.java

```diff
@@ -0,0 +1,80 @@
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.cli.client;
+
+import org.eclipse.hono.cli.AbstractCliClient;
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.springframework.shell.Availability;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellMethod;
+import org.springframework.shell.standard.ShellMethodAvailability;
+import org.springframework.shell.standard.ShellOption;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class contains the methods that the user can execute to create or destroy the factory given connection info.
+ * The vertx instance wil be prepared and the factory correspondent to the profile chosen will be created.
+ */
+@ShellComponent
+@Component
+public class Client extends AbstractCliClient {
+
+    @ShellMethod("Create the connection to a given endpoint\n" +
+            "\t\t\t[-H --host] [-P --port] [-u --username] [-p --password]")
+    void connect(@ShellOption(value = {"-H", "--host"},  defaultValue = "", help = "The address of the endpoint which you want to establish a connection to") final String host,
+                 @ShellOption(value = {"-P", "--port"}, defaultValue = "0", help = "The port used by the service on the connection endpoint") final int port,
+                 @ShellOption(value = {"-u", "--username"}, defaultValue = "", help = "Credentials to use to connect to the endpoint") final String username,
+                 @ShellOption(value = {"-p", "--password"}, defaultValue = "", help = "Credentials to use to connect to the endpoint") final String password) {
+        if (clientConfig.getClientFactory() != null) {
+            shellHelper.printInfo("The previously connection will be destroyed");
+        }
+        final ClientConfig.Profile profile = ClientConfig.Profile.valueOf(inputReader.selectFromList("Profile", "Please enter one of the [] values", ClientConfig.Profiles, true, null).toUpperCase());
+        final ClientConfigProperties cfp = new ClientConfigProperties(this.honoClientConfig);
+        cfp.setHost(host);
+        cfp.setPort(port);
+        cfp.setUsername(username);
+        cfp.setPassword(password);
+        //create the factory with connection info
+        // The #1765 will provide the factory for the AMQP adapter
+        try {
+            if (profile == ClientConfig.Profile.APPLICATION) {
+                clientConfig.createFactory(vertx, cfp, profile);
+            }
```

Comment:
Actually i added this before commit because device commands not require the creation of the factory, but since the method to check the availability `connectionDevAvailability` check anyway the factory value the device methods can't be executed... my bad.
The integration with the AMQP factory PR will fit.
However i'll wait to re-commit the change, because i haven't figured out what's wrong with the build.

Reply:
The failing integration test on Travis does not seem to be related to your changes. We'll need to look into that separately. I have restarted the Travis build for now, expecting it to succeed this time ...